test/es-module/test-esm-loader-mock.mjs

```diff
@@ -1,9 +1,73 @@
-// Flags: --loader ./test/fixtures/es-module-loaders/mock-loader.mjs
+// Flags: --no-warnings
 import '../common/index.mjs';
-import assert from 'assert/strict';
+import * as fixtures from '../common/fixtures.mjs';
+import assert from 'node:assert/strict';
+import { register } from 'node:module';
+import { MessageChannel } from 'node:worker_threads';
 
-// This is provided by test/fixtures/es-module-loaders/mock-loader.mjs
-import mock from 'node:mock';
+const { port1, port2 } = new MessageChannel();
+
+register(fixtures.fileURL('es-module-loaders/mock-loader.mjs'), {
+  parentURL: import.meta.url,
+  data: { port: port2 },
+  transferList: [port2],
+});
+
+/**
+ * This is the Map that saves *all* the mocked URL -> replacement Module
+ * mappings
+ * @type {Map<string, {namespace, listeners}>}
+ */
+globalThis.mockedModules = new Map();
+let mockVersion = 0;
+
+/**
+ * @param {string} resolved an absolute URL HREF string
+ * @param {object} replacementProperties an object to pick properties from
+ *                                       to act as a module namespace
+ * @returns {object} a mutator object that can update the module namespace
+ *                   since we can't do something like old Object.observe
+ */
+function mock(resolved, replacementProperties) {
+  const exportNames = Object.keys(replacementProperties);
+  const namespace = { __proto__: null };
+  /**
+   * @type {Array<(name: string)=>void>} functions to call whenever an
+   *                                     export name is updated
+   */
+  const listeners = [];
+  for (const name of exportNames) {
+    let currentValueForPropertyName = replacementProperties[name];
+    Object.defineProperty(namespace, name, {
+      enumerable: true,
```

Comment:
```suggestion
      __proto__: null,
      enumerable: true,
```

Reply:
This is a test file. Is prototype pollution relevant?

Reply:
It's a best practice to use `__proto__: null` along with `Object.defineProperty`, I don't see a good reason for not doing that; that being said, I don't feel strongly.